migrations/sql/R__add_party_parsing_functions.sql

```diff
@@ -0,0 +1,47 @@
+CREATE OR REPLACE FUNCTION permitee_party_type(party_name varchar) RETURNS varchar AS $$
+DECLARE
+company_keyword_special varchar := '[-!0-9@#$&()`+/\"]|Mining|Mineral|Resources|National|Regional|Energy|Products| and | of |Pacific|Metal|Canada|Canadian|Engineering|Mountain|Lake';
+--case sensitive keyword
+company_keyword_cs  varchar :='Corp|Inc|Expl|Mine|INC|South|North|West';
+--case insensitive keyword
+company_keyword_ci  varchar :='ltd|limited|co.|holdings|Contracting|llp|Consultants|Enterprise|service|city|ulc|Association|Partnership|Trucking|Property|Division|Industries|Developments';
+BEGIN
+    RETURN(
+        SELECT
+            CASE
+                WHEN
+                    party_name ~* company_keyword_special OR
+                    party_name ~ company_keyword_cs OR
+                    party_name ~* company_keyword_ci
+                THEN 'ORG'
+                ELSE 'PER'
+            END
+        );
+END;
+$$ LANGUAGE plpgsql;
+
+
+CREATE OR REPLACE FUNCTION format_permitee_party_name(permitee_name varchar) RETURNS varchar AS $$
```

Comment:
permit**t**e

Reply:
outside the scope of this PR but I notice we maybe have named our permit_permite function incorrectly as well, probably my fault

Reply:
We need to lern how to spel!